numpy/lib/tests/test_index_tricks.py

```diff
@@ -249,6 +249,26 @@ def test_mgrid_size_none_handling(self, start, stop, step, expected):
         assert_equal(grid.size, expected[0])
         assert_equal(grid_small.size, expected[1])
 
+    def test_accepts_npfloating(self):
+        # regression test for #16466
+        grid64 = mgrid[0.1:0.33:0.1, ]
+        grid32 = mgrid[np.float32(0.1):np.float32(0.33):np.float32(0.1), ]
+        assert_(not isinstance(grid32.dtype, int))
```

Comment:
This test is impossible to make fail, `isinstance` isn't useful on dtypes

Reply:
```suggestion
        assert grid32.dtype == np.float32
```

Reply:
whoops, it should be `np.float64` though as `mgrid` only outputs `float64` or `int64` currently.